### PR TITLE
mvcc/reader: fix check lock not return the right ts

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

###  What have you changed?
TiDB uses the max ts to get the latest data in some cases. If there is a primary lock on the key, TiKV will ignore it to get the latest data.

In the original implementation, TiKV will get the latest data with `lock.ts -1` because in the Percolator, the latest committed data must have a smaller `commit_ts`. But in the pessimistic transaction, it's possible the latest `commit_ts` is greater than `lock.ts`.

###  What is the type of the changes?
- Bugfix (a change which fixes an issue)

###  How is the PR tested?
Unit tests and Jepsen tests.

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

File: src/storage/mvcc/reader/util.rs [hunk 1]
```diff
@@ -11,12 +11,8 @@ pub enum CheckLockResult {
     /// Key is locked. The key lock error is included.
     Locked(Error),
 
-    /// Key is not locked.
+    /// Key is not locked or the lock is ignored.
     NotLocked,
-
-    /// Key's lock exists but was ignored because of requesting the latest committed version
-    /// for the primary key. The committed version is included.
-    Ignored(u64),
```
Comment (sticnarf) [inline src/storage/mvcc/reader/util.rs]: Now `CheckLockResult` can be just alias to `Result<Error>` I think.

Comment (youjiali1995) [reply src/storage/mvcc/reader/util.rs]: Yes. I will change it.

**Comment by coocood** [comment]
LGTM

**Comment by youjiali1995** [comment]
@sticnarf PTAL again. Thanks.

**Pull request merged**